traitsui/qt4/list_editor.py

```diff
@@ -311,7 +311,12 @@ def add_item(self, offset):
         index += offset
         item_trait = self._trait_handler.item_trait
         value = item_trait.default_value_for(self.object, self.name)
```

Comment:
This feels like the real problem.  We are relying on the trait to give us the default value, which is great when it works, but it is possible for the default value to _not_ validate for various reasons.  Having this happen is effectively a programmer error outside our control from our point of view - the trait doesn't have a valid default!  So crashing is OK, but it can't be a hard crash.

This PR fixes the hard crash, which is good.

But there is a secondary fix, which is that the buggy code may be outside of the control of the person using the editor, so we need _another_ way of specifying the default value that doesn't rely on the trait.  It feels like the list editor should grow a new "default_factory" `Callable` trait that, if not `None`, is called instead of this.

Reply:
looks like we need to open a separate issue for ^

Reply:
I'v opened #1627